library/std/src/path.rs

```diff
@@ -1724,6 +1747,14 @@ impl ops::Deref for PathBuf {
     }
 }
 
+#[unstable(feature = "path_as_mut_os_str", issue = "none")]
+impl ops::DerefMut for PathBuf {
+    #[inline]
+    fn deref_mut(&mut self) -> &mut Path {
+        Path::from_inner_mut(&mut self.inner)
+    }
+}
```

Comment:
Please move this impl to a separate PR because there is no mechanism to feature-gate DerefMut impls, so it requires team approval to merge as instantly stable.

Does there end up being no possible way to write example code for `as_mut_os_str` without this?

You could try `into_boxed_path` (with a comment that links to the DerefMut PR, and a note in the tracking issue to follow up with more realistic example code after DerefMut is available).